src/classes/RD2_Settings.cls

```diff
@@ -0,0 +1,197 @@
+        get {
+            if (settings == null) {
+                settings = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+            } 
+            return settings;
+        }
+        set;
+    }
+
+    /***
+    * @description The installment Opportunity Stage
+    */
+    public String installmentStage {
+        get {
+            if (installmentStage == null) {
+                installmentStage = String.isNotBlank(settings.InstallmentOppStageName__c)
+                    ? settings.InstallmentOppStageName__c
+                    : System.Label.npe03.RecurringDonationStageName;
+            }
+            return installmentStage;
+        } 
+        private set;
+    }
+
+    /***
+    * @description Installment Opp Auto Creation Option
+    */
+    public String installmentOppAutoCreateOption {
+        get {
+            if (installmentOppAutoCreateOption == null) {
+                installmentOppAutoCreateOption = String.isNotBlank(settings.installmentOppAutoCreateOption__c)
+                    ? settings.installmentOppAutoCreateOption__c
+                    : RD2_Constants.InstallmentCreation.Always_Create_Next_Installment.name();
+            }
+            return installmentOppAutoCreateOption;
+        }
+        private set;
+    }
+
+    /***
+    * @description The installment Opportunity Record Type Id
+    */
+    public Id installmentRecordTypeId {
+        get {            
+            return (Id) settings.npe03__Record_Type__c;
+        } 
+        private set;
+    }
+
+    /***
+    * @description Indicates if Campaign should be propagated from RD to the installment Opportunity
+    */
+    public Boolean shouldSetInstallmentCampaign {
+        get {            
+            return settings.npe03__Add_Campaign_to_All_Opportunites__c;
+        } 
+        private set;
+    }
+
+    /***
+    * @description Indicates if open installment Opportunities should be deleted when RD is closed
+    */
+    public Boolean isCloseActionDeletion() {
+        return settings.npe03__Open_Opportunity_Behavior__c == RD2_Constants.CloseActions.Delete_Open_Opportunities.name();
+    }
+
+    /***
+    * @description Indicates if open installment Opportunities should be set to Closed Lost when RD is closed
+    */
+    public Boolean isCloseActionClosedLost() {
+        return settings.npe03__Open_Opportunity_Behavior__c == RD2_Constants.CloseActions.Mark_Opportunities_Closed_Lost.name();
+    }
+
+    /***
+    * @description Indicates if open installment Opportunities should be unchanged when RD is closed
+    */
+    public Boolean isCloseActionNoAction() {
+        return settings.npe03__Open_Opportunity_Behavior__c == RD2_Constants.CloseActions.No_Action.name();
+    }
+
+    /***
+    * @description Indicates if Installment opps should always be created during RD DML operation
+    */
+    public Boolean isInstallmentSettingAlwaysCreateNextInstallment() {
```

Comment:
The `isFirstInstallmentEnabled` and `isNextInstallmentEnabled` still exist in the settings and they are being used in the Opp eval service but not being populated. Thus, let's remove new methods and reuse existing properties and plug them into Opp processing.
Also, let's use retrieved settings `installmentOppAutoCreateOption` (suggestion to rename to `installmentCreateOption`) instead of verifying the field again. It seems this field is not being used currently, but will be based on the review. Properties are using lazy-load, so the value is set once and reused instead of evaluated every time it is required.
```   
    /***
    * @description Installment Opp Auto Creation Option
    */
    public String installmentCreateOption {
        get {
            if (installmentCreateOption == null) {
                installmentCreateOption = String.isNotBlank(settings.installmentOppAutoCreateOption__c)
                    ? settings.installmentOppAutoCreateOption__c
                    : RD2_Constants.InstallmentCreation.Always_Create_Next_Installment.name();
            }
            return installmentCreateOption;
        }
        private set;
    }
    
    /***
    * @description Indicates if the first installment can be created on a Recurring Donation
    */
    public Boolean isFirstInstallmentEnabled {
        get {
            if (isFirstInstallmentEnabled == null) {
                if (isDisableAllInstallments) {
                    isFirstInstallmentEnabled = false;
                } else {
                    isFirstInstallmentEnabled = 
                        installmentCreateOption != RD2_Constants.InstallmentCreation.Disable_First_Installment.name();
                }
            }
            return isFirstInstallmentEnabled;
        } 
        private set;
    }

    /***
    * @description Indicates if the next installment can be created on a Recurring Donation
    */
    public Boolean isNextInstallmentEnabled {
        get {
            if (isNextInstallmentEnabled == null) {
                isNextInstallmentEnabled = !isDisableAllInstallments;
            }
            return isNextInstallmentEnabled;
        } 
        private set;
    }

    /***
    * @description Indicates if any installment creation on a Recurring Donation is disabled
    */
    public Boolean isDisableAllInstallments {
        get {
            if (isDisableAllInstallments == null) {
                isDisableAllInstallments = 
                    installmentCreateOption == RD2_Constants.InstallmentCreation.Disable_All_Installments.name();
            }
            return isDisableAllInstallments;
        } 
        private set;
    }
```